src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,53 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can`t be null!");
+        }
+
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must contain at least 6 symbols!");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists!");
+        }
+
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must contain at least 6 symbols!");
+        }
+
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new RegistrationException("User`s age must be at least 18 years old!");
+        }
+        return storageDao.add(user);
+    }
+
+    private boolean isPasswordComplex(String password) {
```

Comment:
There is a method `isPasswordComplex` which checks for uppercase, lowercase, digit, and special symbol in the password. However, this method is never used and the task does not require a check for password complexity. Please remove unnecessary code.